### PR TITLE
[Accessibility] Make the edit/delete actions of shipping/products accessible in manual orders

### LINKED ISSUE #26875: [Accessibility] Manual Order Shipping is not accessibility via screen reader
Submitted via WooCommerce Support:

Steps to reproduce:
Go to Add Order Manually
Add Item
Add Shipping

What you get then is a shipping rate of 0.00 and what is described to me by my sighted assistant as a “Pencil” to the right of it. You have to click the pencil to edit the shipping details, such as amount and shipping type.

There is no way to click that pencil via the keyboard. Pressing enter on 0.00 yields nothing and I can’t edit it without assistance.

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #26875 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Orders -> Add Order.
2. Click on the "Add item(s)" button.
3. Click on the "Add shipping" button.
4. Tab navigate through the page and verify that you can focus on both the "Edit shipping" and "Delete shipping" buttons.
image

5. Use a screen reader and verify that "Edit shipping" and "Delete shipping" are announced once the correspondent buttons are focused.
6. Hover over the Shipping line and verify that the "Edit shipping" and "Delete shipping" show up.
7. Hover off the Shipping line and verify that the "Edit shipping" and "Delete shipping" disappear.
8. Add a product by clicking on the "Add product(s)" buttons and repeat the steps above but this time using the "Edit item" and "Delete item" buttons on the Product line.
image
9. Add a fee by clicking on the "Add fee" button and repeat the steps above but this time using the "Edit fee" and "Delete fee" buttons on the Fee line.
image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by Manussakis** [comment]
@dkotter This PR is ready for code review

**Review by dkotter — APPROVED** [review]

**Comment by dkotter** [comment]
@qasumitbagthariya @ankitguptaindia  This is ready for QA

**Comment by ankitguptaindia** [comment]
Thanks, @Manussakis, for the PR. I was able to test it and found a concern that I'd like your feedback on point 2. 

✅  Point:1- On Tab navigation, both buttons have focus now with this PR and  "Edit shipping" and "Delete shipping" are announced when correspondent buttons are focused. 


https://github.com/woocommerce/woocommerce/assets/6384299/b2652eac-97d3-46d1-9858-bff4421a4247





⚠️ Point:2 - However, when hovering over the "Edit shipping" and "Delete shipping" options in the Shipping line, no text is displayed. The text only appears after performing another action and then hovering over the "Edit shipping" and "Delete shipping" icons again. Please watch below attached video for detailed steps. 




https://github.com/woocommerce/woocommerce/assets/6384299/b7cba078-4317-4518-a2a5-4619272404c4


**Testing Environment -** 

<Details>

* WordPress: 6.5.4
* Theme: Storefront Version: 4.5.4 
* WooCommerce - Currect PR branch.
* PHP: 8.3
* Web Server: Nginx (Local hosted site LocalWP)
* Browser: Chrome -Version 125.0.6422.142 
* OS: macOS Sonoma 14.4.1 
* Git Branch: `Manussakis:fix/26875-manual-order-shipping-accessibility` 
* Voice Over: Version 10 


</Details>




**Next Step-** Moving back to feedback (in-progress column).

**Comment by Manussakis** [comment]
@ankitguptaindia Thanks for testing this PR!
Point 2 – Good catch! Fixed.

@dkotter These are the two commits I pushed after your last review:
- [Init tiptip when adding shipping](https://github.com/woocommerce/woocommerce/pull/48238/commits/10486a67adc29137c597b1a81dfc2bc5f11dce44)
- [Remove unnecessary white space](https://github.com/woocommerce/woocommerce/pull/48238/commits/f134543e49584f81fb569e0df646430703ec3428)

Please, re-review it 🙂

**Review by dkotter — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.5.4
* Theme: Storefront Version: 4.5.4 
* WooCommerce - Currect PR branch. (Dev version) 
* PHP: 8.3
* Web Server: Nginx (Local hosted site LocalWP)
* Browser: Chrome -Version 126 
* OS: macOS Sonoma 14.4.1 
* Git Branch: `Manussakis:fix/26875-manual-order-shipping-accessibility` 
* Voice Over: Version 10 


</Details>




**Test Results -**  Reported accessibility issues and QA feedback have been addressed and working as expected now. 


**Functional Demo /  Screencast -**

**Test Result with macOS Inbuilt Voiceover:** 

https://github.com/woocommerce/woocommerce/assets/6384299/d6deb943-2a6f-4e14-86d4-1e16a0a77453

**Test Result with Screen Reader Chrome Extension:** 


https://github.com/woocommerce/woocommerce/assets/6384299/4e92d0fd-81a5-47b9-8463-cbdbe50eeb83

**Comment by jeffpaul** [comment]
@coreymckrill this is now ready for Woo review.

@Manussakis please remove the WIP from the PR title.

**Review by coreymckrill — COMMENTED** [review]
@Manussakis the shipping and product line buttons look good, I see the tooltips for the edit and delete buttons on both, and the voiceover sounds great. I noticed that fee lines also suffer from a similar issue to what was reported in #26875 (no tooltips on the buttons, voiceover just says "comma"). Do you think that would be worth adding in this PR, since it's very similar? Also fine if you'd prefer to address it in a separate PR, since it wasn't in the original scope of the issue.

**Comment by Manussakis** [comment]
> @Manussakis the shipping and product line buttons look good, I see the tooltips for the edit and delete buttons on both, and the voiceover sounds great. I noticed that fee lines also suffer from a similar issue to what was reported in #26875 (no tooltips on the buttons, voiceover just says "comma"). Do you think that would be worth adding in this PR, since it's very similar? Also fine if you'd prefer to address it in a separate PR, since it wasn't in the original scope of the issue.

@coreymckrill Thanks for calling this out! I believe it makes sense to use this PR to fix the buttons on the fee lines.
I will fix it.

**Comment by Manussakis** [comment]
@dkotter , after [this comment](https://github.com/woocommerce/woocommerce/pull/48238#pullrequestreview-2140251848) by @coreymckrill , I took over this PR to update the buttons on the Fee line.
Please review this commit:
[Add tooltips to buttons on the fee line](https://github.com/woocommerce/woocommerce/pull/48238/commits/f547ea5ed8355c4b67293239c8e2e6d0b96af78f)

@ankitguptaindia I believe another round of QA is worth it after Darin's review.

**Review by dkotter — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
QA/Test Report- ✅
---

New changes related to Fee tab have been tested successfully and all tests working as expected. 

Tested with macOS Voice Over and Chrome Screen reader extension. 


https://github.com/woocommerce/woocommerce/assets/6384299/d8d3b3a1-0af5-41fa-abf1-4de7a36c850a

**Comment by vikrampm1** [comment]
@coreymckrill this is back for your review again.

**Comment by jeffpaul** [comment]
@woocommerce/proton note that this PR is ready for your review

File: plugins/woocommerce/client/legacy/css/admin.scss [hunk 1]
```diff
@@ -2698,12 +2698,13 @@ ul.wc_coupon_list_block {
 			color: #ccc;
 			display: inline-block;
 			cursor: pointer;
-			padding: 0 0 0.5em;
-			margin: 0 0 0 12px;
+			padding: 0;
+			margin: 0 0 0.5em 12px;
```
Comment (naman03malhotra) [inline plugins/woocommerce/client/legacy/css/admin.scss]: Can you please explain why did you use `em` and `px` in the same context?

Comment (Manussakis) [reply plugins/woocommerce/client/legacy/css/admin.scss]: @naman03malhotra Before this PR, the icons were aligned with the text by the padding-bottom value (`padding: 0 0 0.5em;`). But this is what the focus ring looked like:
image

To remove the extra space below the icon, I decided to remove the bottom spacing from the `padding` and add it to the `margin` attribute: `margin: 0 0 0.5em 12px`;. 
image

To keep it consistent as it was before the change, I kept the `0.5em` value.

Please let me know your thoughts.

**Review by coreymckrill — APPROVED** [review]
👍 Thanks for adding the Fee line tooltips as well!

File: plugins/woocommerce/client/legacy/css/admin.scss (hunk 1 above)
Comment (naman03malhotra) [reply plugins/woocommerce/client/legacy/css/admin.scss]: Good practice would be to follow a consistent metric throughout, can you check if we can use px in both places?

**Comment by vikrampm1** [comment]
@Manussakis please move this PR out of draft as you address the remaining comments above.

File: plugins/woocommerce/client/legacy/css/admin.scss (hunk 1 above)
Comment (Manussakis) [reply plugins/woocommerce/client/legacy/css/admin.scss]: @naman03malhotra I fixed it under this commit: [Replace em unit with px](https://github.com/woocommerce/woocommerce/pull/48238/commits/09ad03f754d264536ef9950b571e8d632354adb5)

**Comment by vikrampm1** [comment]
This PR is ready for Woo to proceed with the merging.

**Pull request merged**